docs/proposals/20220712-cluster-api-addon-orchestration.md

```diff
@@ -0,0 +1,448 @@
+
+#### Surface Cluster Add-ons Status
+
+ClusterAddonProviders are responsible for installing add-ons on an arbitrary number of workload clusters. As such, it would be beneficial to see the status of an add-on from the management cluster and avoid having to fetch the kubeconfig of the workload cluster.
+
+In the ClusterAddonProvider for Helm, the status of a Helm release can be conveyed through the status field of a HelmReleaseProxy. It includes a ready status, which conveys that the HelmRelease is installed and up to date with the CRD spec and a failure reason which is set if there is an error in reconciliation. Additinoally, the status can include an array of `clusterv1.Conditions` and define new condition types as needed to offer more specific information.
+
+#### Upgrade strategy
+
+Users will be able to upgrade the add-ons across multiple Clusters at once since updating the version in the HelmChartProxy triggers a Helm upgrade for all the Helm releases it manages. Additionally, HelmChartProxy allows users to upgrade their add-ons in a declarative manner.
+
+Note that hooks and automatic triggers for add-on upgrades when a Cluster is upgraded to a new Kubernetes version is out of scope for this iteration. Users will need to determine if an add-on version upgrade is needed, and if so, set a compatible version in the HelmChartProxy.
+
+In future iterations, we plan to handle the upgrade of Cluster add-ons when the workload Cluster is upgraded to a new Kubernetes version by leveraging the new runtime hooks feature.
+
+#### Managing dependencies
+
+Managing dependencies between HelmChartProxies is out of scope for this project. However, Helm charts define their dependencies in the [Chart.yaml file](https://helm.sh/docs/topics/charts/#the-chartyaml-file) and will install them automatically. If a Helm chart has a critical dependency on another chart, it is the chart owner's responsibility to enable it by default.
+
+### Implementation details 
+
+The implementation will include two CRDs and two controllers as follows. As this is a prototype, the specific details regarding the CRD specs are subject to change.
+
+#### HelmChartProxy CRD
+
+The first CRD, HelmChartProxy, serves as the user interface for defining and configuring Helm charts as well as selecting which workload clusters it will be installed on. 
+
+```go
+// HelmChartProxySpec defines the desired state of HelmChartProxy.
+type HelmChartProxySpec struct {
+	// ClusterSelector selects Clusters with a label that matches the specified key/value pair. The Helm chart will be
+	// installed on all selected Clusters. If a Cluster is no longer selected, the Helm release will be uninstalled.
+	ClusterSelector ClusterSelectorLabel `json:"clusterSelector"`
+
+	// ChartName is the name of the Helm chart in the repository.
+	ChartName string `json:"chartName,omitempty"`
```

Comment:
Does chartName adds value over RepoURL?
Also should it be just Name aligning with RepoURL and Version?

Reply:
Helm requires a chartName along with a repoURL since a repo can have multiple charts.